core/src/test/resources/org/apache/calcite/test/RelOptRulesTest.xml

```diff
@@ -466,6 +466,28 @@ LogicalProject(EXPR$0=[1])
     LogicalJoin(condition=[true], joinType=[inner])
       LogicalTableScan(table=[[CATALOG, SALES, EMP]])
       LogicalTableScan(table=[[CATALOG, SALES, DEPT]])
+]]>
+        </Resource>
+    </TestCase>
+    <TestCase name="testNotPushExpression">
+        <Resource name="sql">
+            <![CDATA[select 1 from emp inner join dept
+            on emp.deptno=dept.deptno and emp.ename is not null]]>
+        </Resource>
+        <Resource name="planBefore">
+            <![CDATA[
+LogicalProject(EXPR$0=[1])
+  LogicalJoin(condition=[AND(=($7, $9), IS NOT NULL($1))], joinType=[inner])
+    LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+    LogicalTableScan(table=[[CATALOG, SALES, DEPT]])
```

Comment:
What is the advantage of your proposal? Sometimes pushing down the non-equi join conditions might have benefits. For example, execution count of non-equi condistions may be reduced from rowcount(a)*rowcount(b) to rowcount(a).

Reply:
Sorry, I misunderstood it.